khtools/bloom_filter.py

```diff
@@ -0,0 +1,253 @@
+                continue
+            sequence = encode_peptide(record['sequence'], molecule)
+            try:
+                kmers = kmerize(sequence, peptide_ksize)
+                for kmer in kmers:
+                    # Convert the k-mer into an integer
+                    hashed = hash_murmur(kmer)
+
+                    # .add can take the hashed integer so we can hash the
+                    #  peptide kmer and add it directly
+                    peptide_bloom_filter.add(hashed)
+            except ValueError:
+                # Sequence length is smaller than k-mer size
+                continue
+    return peptide_bloom_filter
+
+
+def make_peptide_set(peptide_fasta, peptide_ksize, molecule):
+    """Create a python set out of peptide sequence k-mers
+
+    For comparing to the bloom filter in storage and performance
+    """
+    peptide_set = set([])
+
+    with screed.open(peptide_fasta) as records:
+        for record in tqdm(records):
+            if '*' in record['sequence']:
+                continue
+            sequence = encode_peptide(record['sequence'], molecule)
+            try:
+                kmers = kmerize(sequence, peptide_ksize)
+                peptide_set.update(kmers)
+            except ValueError:
+                # Sequence length is smaller than k-mer size
+                continue
+    return peptide_set
+
+
+def maybe_make_peptide_bloom_filter(peptides, peptide_ksize, molecule,
+                                    peptides_are_bloom_filter,
+                                    n_tables=DEFAULT_N_TABLES,
+                                    tablesize=DEFAULT_MAX_TABLESIZE):
+    if peptides_are_bloom_filter:
+        click.echo(
+            f"Loading existing bloom filter from {peptides} and "
+            f"making sure the ksizes match",
+            err=True)
+        peptide_bloom_filter = load_nodegraph(peptides)
+        if peptide_ksize is not None:
+            try:
+                assert peptide_ksize == peptide_bloom_filter.ksize()
+            except AssertionError:
+                raise ValueError(f"Given peptide ksize ({peptide_ksize}) and "
+                                 f"ksize found in bloom filter "
+                                 f"({peptide_bloom_filter.ksize()}) are not"
+                                 f"equal")
+    else:
+        peptide_ksize = get_peptide_ksize(molecule, peptide_ksize)
+        click.echo(
+            f"Creating peptide bloom filter with file: {peptides}\n"
+            f"Using ksize: {peptide_ksize} and molecule: {molecule} "
+            f"...",
+            err=True)
+        peptide_bloom_filter = make_peptide_bloom_filter(
+            peptides, peptide_ksize, molecule=molecule,
+            n_tables=n_tables, tablesize=tablesize)
+    return peptide_bloom_filter
+
+
+def maybe_save_peptide_bloom_filter(peptides, peptide_bloom_filter, molecule,
+                                    save_peptide_bloom_filter):
+    if save_peptide_bloom_filter:
```

Comment:
I'm confused why you would need this if statement. I think it would be more fluid if you created a bloom filter class and had a `save_bloom_filter` method. otherwise forget the if and just save the bloom filter? Are there any cases where you would call this function and not save the bloom filter?